release.nix

```diff
@@ -1,7 +1,37 @@
 let
-  shared = import ./nix/shared.nix { };
+  hsPkgs = import ./default.nix { };
 in
-  { inherit (shared)
-      hasktorch_cpu
-    ;
-  }
+  {
+  	libtorch = hsPkgs.libtorch-ffi.components.all;
+
+  	hasktorch = hsPkgs.hasktorch.components.all;
+
+  	all-examples = hsPkgs.examples.components.all;
+
+  	alexNet = hsPkgs.examples.components.exes.alexNet;
+	autograd = hsPkgs.examples.components.exes.autograd;
+	distill = hsPkgs.examples.components.exes.distill;
+	gaussian-process = hsPkgs.examples.components.exes.gaussian-process;
+	gd-field = hsPkgs.examples.components.exes.gd-field;
+	image-processing = hsPkgs.examples.components.exes.image-processing;
+	load-torchscript = hsPkgs.examples.components.exes.load-torchscript;
+	matrix-factorization = hsPkgs.examples.components.exes.matrix-factorization;
+	minimal-text-example = hsPkgs.examples.components.exes.minimal-text-example;
+	mnist-mlp = hsPkgs.examples.components.exes.mnist-mlp;
+	optimizers = hsPkgs.examples.components.exes.optimizers;
+	regression = hsPkgs.examples.components.exes.regression;
+	rnn = hsPkgs.examples.components.exes.rnn;
+	serialization = hsPkgs.examples.components.exes.serialization;
+	static-mnist = hsPkgs.examples.components.sublibs.static-mnist;
+	static-mnist-cnn = hsPkgs.examples.components.exes.static-mnist-cnn;
+	static-mnist-mlp = hsPkgs.examples.components.exes.static-mnist-mlp;
+	static-xor-mlp = hsPkgs.examples.components.exes.static-xor-mlp;
+	typed-transformer = hsPkgs.examples.components.exes.typed-transformer;
+	vae = hsPkgs.examples.components.exes.vae;
+	xor-mlp = hsPkgs.examples.components.exes.xor-mlp;
+
+	all-experimental = hsPkgs.experimental.components.all;
+
+  	dataloader-cifar10 = hsPkgs.experimental.components.exes.dataloader-cifar10;
+  	untyped-nlp = hsPkgs.experimental.components.exes.untyped-nlp;
```

Comment:
There is a helper function in nixpkgs that should allow you to pull most of these out at once:

https://github.com/input-output-hk/haskell.nix/blob/15fc6b3774774d3b457dfe03006c0c0bdae6d7ad/lib/default.nix#L98
https://github.com/input-output-hk/haskell.nix/blob/15fc6b3774774d3b457dfe03006c0c0bdae6d7ad/lib/default.nix#L55

Basically, you pass `hsPkgs` to `selectLocalPackages`, and then call something like `getAllComponents` or `collectComponents`.

Reply:
ah, I was hoping that something like this exists, but I couldn't find it. thanks so much for the pointer!

Reply:
so I removed `release.nix` entirely, didn't see the point for it.

I used `collectComponents` in `default.nix`. Let me know what you think :)

Reply:
I think this looks good.

If I were to make a suggestion, it would be to take out one level of indirection for some of these top level attributes.

For instance, as it is now, if I want to build the `mnist-mlp` example, I have to do the following:

```console
$ nix build -L examples.examples.mnist-mlp -f ./default.nix
```

It would probably be slightly easier to only have to type `nix build -L examples.mnist-mlp -f ./default.nix`.  (Although this is only a minor nitpick)